network/network_linux.go

```diff
@@ -231,14 +231,25 @@ func readDnsInfo(ifName string) (DNSInfo, error) {
 		return dnsInfo, fmt.Errorf("[net] Console output doesn't have any lines")
 	}
 
+	dnsServerFound := false
 	for _, line := range lineArr {
 		if strings.Contains(line, dnsServersStr) {
```

Comment:
There are two different kind of custom DNS server list which users can add in Azure refer #713 .  I believe we are getting only the eth0 (or whichever interface) and adding them to azure0, do we expect global DNS servers to still work fine ? since they Global would not worry about change in interface properties,

Reply:
not sure how to configure global dnsserver in portal. If it's not tied to nic, then we dont need to worry about that.

Reply:
i veriifed updating vnet custom dnsserver list and it worked

Reply:
Can you add a comment on the missing scenario, where if the NIC DNS server list is updated after the VM and CNI is setup then the VM will need a reboot for them to apply. VNET level (global) DNS servers should get updated even after the VM and CNI are setup i guess (but it will be good to verify).

Reply:
if dns server is updated, vm needs a reboot in any case. It doesn't matter if cni is setup or not. I will add a comment though